ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,488 @@
+    job_span.set_tag_str("component", "ray")
+    job_span.set_tag_str("ray.submission_id", submission_id)
+    tracer.context_provider.activate(job_span)
+    start_long_running_job(job_span)
+
+    try:
+        with tracer.trace("ray.job.submit", service=submission_id, span_type=SpanTypes.RAY) as submit_span:
+            submit_span.set_tag_str("component", "ray")
+            submit_span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
+
+            # Inject the context of the job so that ray.job.run is its child
+            env_vars = kwargs.setdefault("runtime_env", {}).setdefault("env_vars", {})
+            _TraceContext._inject(job_span.context, env_vars)
+            env_vars[RAY_SUBMISSION_ID] = submission_id
+
+            try:
+                resp = wrapped(*args, **kwargs)
+                submit_span.set_tag_str("ray.job.submit_status", "success")
+                return resp
+            except Exception:
+                submit_span.set_tag_str("ray.job.submit_status", "error")
+                raise
+    except Exception as e:
+        job_span.set_tag_str("ray.job.status", "error")
+        job_span.error = 1
+        job_span.set_exc_info(type(e), e, e.__traceback__)
+        job_span.finish()
+        raise e
+
+
+def traced_actor_method_call(wrapped, instance, args, kwargs):
+    """Trace actor method submission, i.e the Actor.func.remote()
+    call
+    """
+    if not tracer:
+        return wrapped(*args, **kwargs)
+
+    actor_name = instance._ray_actor_creation_function_descriptor.class_name
+    method_name = args[0]
+
+    # if _dd_trace_ctx was not injected in the param of the function, it means
+    # we do not want to trace this function, for example: JobSupervisor.ping
+    if not any(p.name == "_dd_trace_ctx" for p in instance._ray_method_signatures[method_name]):
+        return wrapped(*args, **kwargs)
+
+    if tracer.current_span() is None:
+        tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+    with tracer.trace(
+        f"{actor_name}.{method_name}.remote()", service=os.environ.get(RAY_SUBMISSION_ID), span_type=SpanTypes.RAY
+    ) as span:
+        span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
+        set_maybe_big_tag(span, "ray.actor_method.args", kwargs.get("args", {}))
+        set_maybe_big_tag(span, "ray.actor_method.kwargs", kwargs.get("kwargs", {}))
+        _inject_ray_span_tags(span)
+
+        try:
+            _inject_context_in_kwargs(span.context, kwargs)
+            return wrapped(*args, **kwargs)
+        except Exception as e:
+            raise e
+
+
+def traced_wait(wrapped, instance, args, kwargs):
+    """
+    Trace the calls of ray.wait
+    """
+    if not tracer:
+        return wrapped(*args, **kwargs)
+
+    if tracer.current_span() is None:
+        tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+    with long_running_ray_span(
+        "ray.wait",
+        service=os.environ.get("_RAY_SUBMISSION_ID"),
```

Comment:
In `ddtrace/contrib/internal/ray/patch.py`, there's a line that sets a constant called: `RAY_SUBMISSION_ID = "_RAY_SUBMISSION_ID"`. If this constant would be useful in a few places, like in this line, maybe it's useful to throw this in a constants file and have one reference to it.